docs/development-instructions.rst

```diff
@@ -1,30 +1,139 @@
 Development Instructions
-===========================
+=========================
 
-1. Clone the `ArduinoCore-API <https://github.com/arduino/ArduinoCore-API>`_ repo to any convenient location::
+..
+   TODOS:
+   - Contribution guidelines (move here from Contributing.md)
+   - Code conventions
+   - ....
+   - Tools installation:
+      - uncrustify
+      - python ?
+      - pip ?
+      - pre-commit hook?
+      - spellchecker
+      - ... 
+   - Installation the Arduino PSoC6 core
+   - Setting up the development environment
 
-   git clone https://github.com/arduino/ArduinoCore-API.git
+Environment setup
+------------------
 
-2. Locate in your computer where the Arduino15 packages are installed. 
-   e.g. Windows: c:/Users/%USERNAME%/AppData/local/Arduino15
-        Linux:   ~/.arduino15/
+.. note::
+   | The development environment setup is (currently) only supported on **Linux** |:penguin:|. 
+   | As many tools and scripts are not cross-platform, some of then will not directly work on Windows |:abcd:| or macOS |:apple:|.
 
-3. Change directories::
 
-   cd packages/Infineon-psoc/hardware/psoc/0.0.1
+0. Obviously |:neutral_face:|, install `Arduino IDE (2.0 or higher) <https://docs.arduino.cc/software/ide-v2/tutorials/getting-started/ide-v2-downloading-and-installing/>`_ or `Arduino CLI (1.0.0 or higher) <https://arduino.github.io/arduino-cli/0.24/installation/>`_.
 
-4. Clone this repo::
+1. Install the :ref:`Infineon PSoC™ microcontroller <psoc_core_installation>` boards package.
 
-   git clone https://github.com/Infineon/arduino-core-psoc.git
+..
+   Comment: there is another option to this, which is the manual setup:
+      - Creating the "infineon/hardware/psoc" manually
+      - Creating the "infineon/tools" and manually copy there the MTB tools pacakages.
 
-5. Change directories::
+   Ramya: I think we can skip this one, as the contributors should start with the .json package index installation available or?
 
-   cd cores/psoc
+2. Locate in your computer where the Arduino15 packages are installed. Usually:
 
-6. Copy or symlink the ``api`` folder from the `ArduinoCore-API <https://github.com/arduino/ArduinoCore-API>`_ repo::
+   ::
 
-   ln -s <ARDUINO_CORE_API>/api .
+      ~/.arduino15/
+
+ .. TODO: I would postpone the Windows part, as we won´t have all the dev tools available there
+         c:/Users/%USERNAME%/AppData/local/Arduino15 
+
+3. From the Arduino installation location, change directories
+
+   ::
+
+      cd packages/Infineon-psoc/hardware/psoc
+
+   ..
+      TODO: We already have the "Infineon" package (from XMC-for-Arduino). We should go for "infineon" instead of "infineon-psoc". And
```

Comment:
Yes I was also thinking that. But the problem is when both xmc and psoc needs to be installed, we should check if the package index json will get replaced or extended.

Reply:
Yes, we have to check that. Arduino has cores in different repos (and they all end up under /packages/arduino, but I think only one json file. I guess it should work if splitted in different jsons?
Easy to check anyhow :)

Reply:
Yes I am preparing a new release and once we have it we can check this